android/src/main/java/com/reactnativereceivesharingintent/RNMediaInfoUtils.java

```diff
@@ -0,0 +1,197 @@
+                    String height = exif.getAttribute(ExifInterface.TAG_IMAGE_LENGTH);
+
+                    if (width != null && height != null ) {
+                        resultMap.put("width",width);
+                        resultMap.put("height",height);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            return resultMap;
+        } catch (Exception e) {
+            e.printStackTrace();
+            Log.e(TAG, "Error getting media dimensions", e);
+            return new HashMap<>();
+        }
+    }
+
+    // TODO: Expand with MediaMetadataRetriever for video metadata
+    // https://developer.android.com/reference/android/media/MediaMetadataRetriever
+    public static WritableMap getCompleteMediaInfo(Context context, Uri contentUri, String subject) {
+        Log.d(TAG, "Starting getCompleteMediaInfo with URI: " + contentUri.toString() + " and subject: " + subject);
+
+        WritableMap completeInfoMap = new WritableNativeMap();;
+        WritableMap exifMap = new WritableNativeMap();;
+        try {
+            String absolutePath = getAbsolutePath(context, contentUri);
+            completeInfoMap.putString("filePath", absolutePath);
+            completeInfoMap.putString("contentUri", contentUri.toString());
+
+            // Other string mappings
+            completeInfoMap.putString("text", null);
+            completeInfoMap.putString("weblink", null);
+            completeInfoMap.putString("subject", subject);
+
+            String mimeType = context.getContentResolver().getType(contentUri);
+
+            if (mimeType != null) {
+                completeInfoMap.putString("mimeType", mimeType);
+            }
+
+            Boolean isImage = mimeType != null && mimeType.startsWith("image/");
+
+            // Fetch Media Info
+            Map<String, String> mediaInfo = getMediaInfo(context, contentUri);
+            if (mediaInfo.containsKey("fileName")) {
+                completeInfoMap.putString("fileName", mediaInfo.get("fileName"));
+            }
+            if (mediaInfo.containsKey("fileSize")) {
+                completeInfoMap.putString("fileSize", mediaInfo.get("fileSize"));
+            }
+            if (mediaInfo.containsKey("DateTimeTaken")) {
+                exifMap.putString("DateTimeTaken", mediaInfo.get("DateTimeTaken"));
+            }
+            if (mediaInfo.containsKey("DateTimeModified")) {
+                exifMap.putString("DateTimeModified", mediaInfo.get("DateTimeModified"));
+            }
+
+            // Fetch Media Dimensions
+            Map<String, String> dimensions = getMediaDimensions(context, contentUri, isImage);
+            if (dimensions.containsKey("width")) {
+                exifMap.putString("width", dimensions.get("width"));
+            }
+            if (dimensions.containsKey("height")) {
+                exifMap.putString("height", dimensions.get("height"));
```

Comment:
Minor suggestion for better readability - could getMediaDimensions always return width and height (even if undefined/empty) so these two ifs here aren't needed?
Not sure if that would cause problems or cover all cases, but it would be a bit easier to read

Reply:
I definitely agree that this code has room for improvement regarding readability. I would like to focus on that on next iteration, where will build this from scratch. 

Currently I can't really asses the outcome if we do any changes.